### ISSUE TITLE
Errata

### ISSUE THREAD

**Issue by Myndex** [xi/apca-introduction#2]
I am quoting the "Missing Introduction" below, indicating misunderstandings and errors:

> _APCA was created by Andrew Somers (Myndex) and is currently being proposed for the next major version of the W3C Accessibility Guidelines_

As well as for use in other standards and contexts.

> _An interactive demo is available at <https://xi.github.io/apca-introduction/tool/>._

Your demo as linked here is not _fully_ compliant with APCA. Please see: https://git.apcacontrast.com/documentation/minimum_compliance

> - _WCAG 2.x produces a ratio between 1:1 and 21:1. APCA produces a value roughly between -100 and 100._

More correct: APCA produces a value of 0 to 106 for dark text on a lighter background, and 0 to -108 for light text on a darker background.

> - _The result of APCA is negative for light text on dark background. You will usually work with the absolute value though._

Developers have the option of returning a signed value to indicate polarity, OR returning a value with a string identifying the polarity. I.e.:

Signed value:  75  or -75
Text Ident:    75 BoW or 75 WoB
Text Ident:    75 LM or 75 DM
etc.

> - _With WCAG 2.x, 29% of all color combinations meet at least level A, 14% meet at least level AA, and 5% meet level AAA. With APCA, only 23% of all color combinations meet at least level A, 11% meet at least level AA, and 3% meet level AAA. So APCA is stricter_.

There is no contrast requirement for level A, so I'm not sure what you are referring to. Also, how did you derive these numbers? They don't seem correct?

> ## _Examples_

Your examples as chosen are all fairly high contrast using a bold font. This does NOTHING to demonstrate the differences. Your examples are mostly in the realm of "contrast constancy", and not near the edge.

> _Evaluating a contrast algorithm is extremly difficult because contrast perception varies from person to person and also depends on the lighing conditions._

No not really, though there are _contrast sensitivity impairments_ that do have an effect. 

Ambient lighting affects contrast perception in that ambient is part of the driver of light adaptation, and the adapted level affects contrast perception, as does context. All of these issues though are part of the contrast matching experiments that instructed the curve shaping that was done in developing APCA, and are set to the "lowest common worst case".

> _Whether APCA is actually better than WCAG 2.x is therefore hard to tell_.

Actually, it is prima facie evidence, and trivial to demonstrate. Here's an example:

<img alt="ColumnCompareAll400" width="540" src="https://user-images.githubusercontent.com/42009457/181414695-1e6e7c19-d10b-4d3c-a360-51bffff668e5.png">

And here is a comparison for dark mode:

<img alt="ColumnCompareAll400" width="540" src="https://user-images.githubusercontent.com/42009457/181415022-dc1b7fa1-0b32-4956-a3a5-589460811333.png">

> _I personally could not say from the examples above which one works better for me._

Your examples are poor, using a bold font, and using high contrast, the result is that your examples are above contrast constancy, and therefore do not demonstrate the important differences.

> _A rigorous scientific evaluation is not yet available._

??? There has been ample third party and peer review. [Here are just a couple](https://git.myndex.com/#apca-peer-review--third-party-discussion)

It is true there are uninformed claiming otherwise, and they fully ignore the reviews that have been completed. Also, ACPA is the result of three years of development in the Visual Contrast subgroup of Silver, and under the oversight of the AGWG.

> _For one it was born out of my personal frustration with the original
documentation. Some important pieces of information (e.g. the actual algorithm)
get buried under all that text._

LOL. You never bothered to look in the folder labeled "documentation", the first file in the list is the algorithm. Not to mention that the JS file shows the algorithm plain as day, along with ample comments.

> _The original documentation also contains absolute statements like "APCA is perceptually uniform" and that the old algorithm produces "invalid results". This in my opinion is wrong as perceptual uniformity is an ideal that can never be reached completely. So I felt like there was room for a more balanced introduction._

You are by your own admission not a vision scientist. "Perceptually uniform" has a specific meaning in the context of the field, namely that that the delta value matches the perceived delta.

WCAG 2 contrast math is nowhere close to perceptual uniformity, as is plain to see in the above examples. 

Perceptual uniformity means, in this context, that a lightest pair of colors at Lc 45 is just as readable as a darkest pair of colors also at the wsame Lc 45.

> _If you want to dig deeper, I recommend to start with the [original WCAG issue](https://github.com/w3c/wcag/issues/695) and the [documentation README](https://git.apcacontrast.com/documentation/README)._

Thread 695 is over three years old, and is _**not**_ a good place to start. The documentation readme you linked to is fine, but the catalog of resources is https://git.myndex.com

Thank you for reading.

**Comment by xi** [comment]
Thanks for the comments! I think it is valuable to get your position on these claims.

> There is no contrast requirement for level A, so I'm not sure what you are referring to.

Yes, this was misleading. I was using `WCAG 7:1 = APCA 75 = AAA` etc, but that is just not correct. Should now be fixed in 1c287a2.

> Also, how did you derive these numbers? They don't seem correct?

The numbers were from an older version of the [coverage script](https://github.com/xi/apca-introduction/blob/main/plots/coverage.py). I aligned them with the current script in b39fb7a. So now they also match the numbers from analysis.md.

> Examples

Thanks for the feedback! I will have to look into that. I opened the separate issue #5 for more focus.

>> Whether APCA is actually better than WCAG 2.x is therefore hard to tell.
>
> Actually, it is prima facie evidence, and trivial to demonstrate. Here's an example:

The images you posted are far from obvious evidence, at least for me. I am not sure why you treat them as such.

>> A rigorous scientific evaluation is not yet available.
>
> ??? There has been ample third party and peer review. [Here are just a couple](https://git.myndex.com/#apca-peer-review--third-party-discussion)

As others have already said, none of that qualifies as scientific evaluation or peer review. So I will stand by that sentence.

Thanks again for your comments and also for the work on APCA. I learned a lot from it already and hope my contributions are helpful.

**Comment by Myndex** [comment]
> _As others have already said, none of that qualifies as scientific evaluation or peer review. So I will stand by that sentence._

The "others" don't understand terms such as "efficacy" and "peer review" and are applying the narrow FDA meaning which is notwithstanding, not the scientific meaning. APCA is an practical application of peer reviewed, scientific consensus CAM models and readability research, as detailed in the bibliography. Claims otherwise are nothing more than trolling.
